server-base.R

```diff
@@ -68,11 +71,39 @@ shinyServer(function(input, output, session){
     to_plot$r_quiet@data <<- cbind(to_plot$r_quiet@data[!(names(to_plot$r_quiet) %in% names(to_plot$l))], to_plot$l@data)
     # Add rqincr column to the quiet data
     to_plot$r_quiet@data$rqincr <<- to_plot$r_quiet@data$length / to_plot$r_fast@data$length
+    region$repopulate_region <<- F
+    #isolate(region$replot <- !region$replot)
+  })
+
+  region <- reactiveValues(current = starting_city, data_dir = file.path(data_dir_root, starting_city), repopulate_region = F) # replot = F,
 
-    isolate(region$replot <- !region$replot)
+  observe({
+    # If a region does not have an 'all-trips'directory, disable the dropdown menu
+    if (!dir.exists(file.path(data_dir_root, starting_city, 'all-trips'))){
+      shinyjs::disable("trip_type")
+      # hide trip_menu
+      shinyjs::hide("trip_menu")
+    }
   })
 
-  region <- reactiveValues(current = starting_city, data_dir = file.path(data_dir_root, starting_city), replot = F )
+
+  observe({
+    # Create a reactive expression on the type of trips dropdown menu
+    input$trip_type
+
+    # Check if the data folder of a specific region contains a subfolder called 'all-trip'
+    # If it does, only then load 'all-trip' data or load defaul commute data
+    if (dir.exists(file.path(data_dir_root, starting_city, 'all-trips'))){
+      if (input$trip_type == 'All'){
+        region$data_dir <<- file.path(data_dir_root, starting_city, 'all-trips')
+      }
+      else{
+        region$data_dir <<- file.path(data_dir_root, starting_city)
+      }
+      # redraw_zones()
+      region$repopulate_region <<- T
```

Comment:
Does this need to be set to false when you change region?  Not an issue now as GM is isolated.